operators/advanced_network/python/adv_network_common_pybind.cpp

```diff
@@ -17,109 +17,316 @@
-        "Free all packets in a burst for one segment");
-  m.def("free_all_packets_and_burst_rx",
-        py::overload_cast<BurstParams*>(&free_all_packets_and_burst_rx),
-        "Free all packets and burst structure for RX");
-  m.def("free_all_packets_and_burst_tx",
-        py::overload_cast<BurstParams*>(&free_all_packets_and_burst_tx),
-        "Free all packets and burst structure for TX");
-  m.def("free_segment_packets_and_burst",
-        py::overload_cast<BurstParams*, int>(&free_segment_packets_and_burst),
-        "Free all packets and burst structure for one packet segment");
+        "Get length of the packet",
+        "burst"_a,
+        "idx"_a);
+  m.def("get_packet_flow_id",
+        py::overload_cast<BurstParams*, int>(&get_packet_flow_id),
+        "Get flow ID of a packet",
+        "burst"_a,
+        "idx"_a);
+
+  // Packet pointer functions
+  m.def("get_segment_packet_ptr",
+        py::overload_cast<BurstParams*, int, int>(&get_segment_packet_ptr),
+        "Get packet pointer for one segment",
+        "burst"_a,
+        "seg"_a,
+        "idx"_a);
+  m.def("get_packet_ptr",
+        py::overload_cast<BurstParams*, int>(&get_packet_ptr),
+        "Get packet pointer",
+        "burst"_a,
+        "idx"_a);
+
+  // TX burst functions
   m.def("tx_burst_available",
         py::overload_cast<BurstParams*>(&is_tx_burst_available),
-        "Return true if a TX burst is available for use");
+        "Return true if a TX burst is available for use",
+        "burst"_a);
   m.def("get_tx_packet_burst",
         py::overload_cast<BurstParams*>(&get_tx_packet_burst),
-        "Get TX packet burst");
-  m.def("shutdown", (&shutdown), "Shut down the advanced_network manager");
-  m.def("print_stats", (&print_stats), "Print statistics from the advanced_network manager");
-  // m.def("set_cpu_udp_payload",
-  //     [](BurstParams *burst, int idx, long int data, int len) {
-  //             return set_cpu_udp_payload(burst, idx,
-  //                     reinterpret_cast<void*>(data), len); },
-  //             "Set UDP header parameters and copy payload");
-  // m.def("set_cpu_udp_payload",
-  //     [](std::shared_ptr<BurstParams> burst, int idx, long int data, int len) {
-  //         return set_cpu_udp_payload(burst, idx,
-  //              reinterpret_cast<void*>(data), len); },
-  //         "Set UDP header parameters and copy payload");
+        "Get TX packet burst",
+        "burst"_a);
+  m.def("send_tx_burst",
+        py::overload_cast<BurstParams*>(&send_tx_burst),
+        "Send a TX burst",
+        "burst"_a);
+
+  // RX burst functions - all overloads
+  m.def(
+      "get_rx_burst",
+      [](int port, int q) {
+        BurstParams* burst_ptr = nullptr;
+        Status status = get_rx_burst(&burst_ptr, port, q);
+        return py::make_tuple(status, py::cast(burst_ptr, py::return_value_policy::reference));
+      },
+      "Get RX burst for specific port and queue",
```

Comment:
I think this is okay, but the user is responsible for calling `free_rx_burst` from Python to free the object when done with it? Since Python authors are likely not used to explicitly freeing resources, we may need to mention that in the docstring here (and in the other overloads for this function just below)